spanner_orm/admin/column.py

```diff
@@ -39,8 +41,36 @@ def nullable(self) -> bool:
 
   def field_type(self) -> Type[field.FieldType]:
     for field_type in field.ALL_TYPES:
-      if self.spanner_type == field_type.ddl():
+      len = _get_type_len(self.spanner_type)
+      contain_commit_timestamp_option = _get_commit_timestamp_option(
+          self.spanner_type)
+      actual_type = field_type.ddl()
+      if len:
+        actual_type = actual_type.replace('(MAX)', f'({len})')
+      if contain_commit_timestamp_option:
+        actual_type = f'{actual_type}{allow_commit_timestamp_option}'
+      if self.spanner_type == actual_type:
         return field_type
 
     raise error.SpannerError('No corresponding Type for {}'.format(
         self.spanner_type))
+
+
+def _get_type_len(spanner_type: str) -> int:
+  """Retrieve length for existing STRING, BYTES or ARRAY<STRING> field."""
+  bytes_len = spanner_type[6:-1]
+  str_len = spanner_type[7:-1]
+  array_str_len = spanner_type[13:-2]
```

Comment:
How is this not crashing with index out of bounds?

Reply:
Whenever out of bounds, it simply returns an empty string